### DISCUSSION TITLE
Fixup Workflow

### DISCUSSION THREAD

**Discussion by chukinas** [tpope/vim-fugitive#2066] [Q&A] [answered]
I use git fixup commits all the time and would love to figure out how best to leverage Fugitive.

There are the `cf` and `cF` commit maps. Those work great if I have unpushed commits. All I have to do is open the Fugitive window with `:G<CR>`, commit the desired changes, then cursor down to the commit I wanna fix up, and `cf`.

How do I do this though if I've pushed all my commits already and they're not showing up by default?

**Comment by tpope** [comment, marked as answer]
I recommend keeping your tracking branch as the remote branch you're intending to merge to (typically `main` or `master`), _not_ the remote branch you're pushing to. This can be slightly clumsy to achieve with the Git command line, so I typically use an alias to create new branches:

```gitconfig
[alias]
        work = !sh -c 'git fetch && git checkout @{upstream} -t -b \"$@\"' _
[push]
        default = current
[pull]
        rebase = true
```

**Comment by marlun** [comment]
Can you explain some more how you're supposed to use the work alias? Let's say I've got a main branch and a foobar branch where I'm working on  a PR where I've pushed all commits so they're not shown in :G

**Reply by tpope** [reply]
You use it to _create_ and check out the foobar branch. If you're already created the branch, you can change said branch to have an `origin` remote and `main` head instead.

**Reply by marlun** [reply]
I managed to use it to create a new branch hand have it's upstream be `origin/main` and by having `push.default` be "current" get the state to what you recommend which lets me see all commits in the status window.

However, I'm not sure I understand *change said branch to have an origin remote and main head* for an already created branch, but I think I managed to get to the same state for an existing branch by just setting `origin/main` as upstream with: `git branch -u origin/main`

**Reply by tpope** [reply]
`git branch -u origin/main` sounds right.
